breakpad-symbols/src/sym_file/types.rs

```diff
@@ -45,18 +52,34 @@ pub struct Function {
     /// The name of the function as declared in the source.
     pub name: String,
     /// Source line information for this function.
-    pub lines: RangeMap<u64, SourceLine>,
+    pub lines: Vec<SourceLine>,
 }
 
 impl Function {
+    pub fn end_address(&self) -> u64 {
+        self.address.saturating_add(self.size.into())
+    }
+
     pub fn memory_range(&self) -> Option<Range<u64>> {
         if self.size == 0 {
             return None;
         }
-        Some(Range::new(
-            self.address,
-            self.address.checked_add(self.size as u64)? - 1,
-        ))
+        Some(self.address..self.end_address())
+    }
+
+    pub fn line_for_address(&self, address: u64) -> Option<&SourceLine> {
+        match self.lines.binary_search_by_key(&address, |l| l.address) {
```

Comment:
I would prefer if you could transplant the comment you deleted, binary-search is hell and always merits comments.

(edit: oh wait you added some to the other instance, this probably still merits them though)

Reply:
ok, will make sure there are some comments at every instance of this pattern